### PR TITLE
Make `stack-usage` display inline, same as `opt-view`

### PR DESCRIPTION
**Pull request by OfekShilon**
Following the intention declaration [here](https://github.com/compiler-explorer/compiler-explorer/pull/6393#issuecomment-2076787747):

image

### CODE REVIEW

File: static/styles/themes/dark-theme.scss [hunk 1]
```diff
@@ -465,18 +465,17 @@ textarea.form-control {
     color: black;
 }
 
-.opt-decoration {
-    &.analysis,
-    &.mixed {
-        background: #fdfd96 !important;
+.stack-usage {
```
Comment (RubenRBS) [inline static/styles/themes/dark-theme.scss:468]: Random thought: Should we give this some ofther font?
Maybe so it looks like a codelens text? https://microsoft.github.io/monaco-editor/playground.html?source=v0.48.0#example-extending-language-services-codelens-provider-example

**Review by RubenRBS — COMMENTED** [review]
Looks great :)

File: static/styles/themes/dark-theme.scss (hunk 1 above)
Comment (OfekShilon) [reply static/styles/themes/dark-theme.scss:468]: Interesting! Does this look better?
image


For opt-remarks I would be interested in making part of some comments text link to source, and I'm not sure if that is possible with InjectedText. Does codeLensProvider enable this?

Comment (mattgodbolt) [reply static/styles/themes/dark-theme.scss:468]: Ooh I prefer the font there? But also I'm cool with merging this and improving/tweaking later :)

**Review by mattgodbolt — APPROVED** [review]
LGTM! thank you Ofek!

File: static/styles/themes/dark-theme.scss (hunk 1 above)
Comment (OfekShilon) [reply static/styles/themes/dark-theme.scss:468]: @mattgodbolt sorry, I don't understand: which font do you prefer?

**Comment by partouf** [comment]
> ## Suspect Issues
> This pull request was deployed and Sentry observed the following issues:
> 
> * ‼️ **TypeError: Cannot read properties of undefined (reading 'replace')** `forEach(panes/stack-usage-view)` [View Issue](https://compiler-explorer.sentry.io/issues/5374209222/?referrer=github-pr-bot)
> 
> Did you find this useful? React with a 👍 or 👎

Never seen this kind of sentry comment, very interesting.

I think if we did `|| 'None'` here https://github.com/compiler-explorer/compiler-explorer/blob/9cdd3a049f6a0d5511879012c2278eb2361a536e/static/panes/stack-usage-view.ts#L156 the problem would be fixed

**Comment by partouf** [comment]
Perhaps also add `?` to the type here https://github.com/compiler-explorer/compiler-explorer/blob/9cdd3a049f6a0d5511879012c2278eb2361a536e/static/panes/stack-usage-view.interfaces.ts#L40

**Comment by OfekShilon** [comment]
I wonder which Qualifier this code misses. I'll also add some sentry message for it.

**Comment by OfekShilon** [comment]
Just pushed https://github.com/compiler-explorer/compiler-explorer/commit/930ee1d4fc13a33688ae3093befd881aa0276957 , in hope of learning more before applying a solution

**Pull request merged**